scripts/update_tags_per_languages.py

```diff
@@ -0,0 +1,365 @@
+    already_referenced_tags = []
+
+    for tag in all_tags["tags"]:
+        if tag["known"] == 0:
+            # limit number of iterations
+            if dev and len(possible_wrong_language_tags) > 0:
+                break
+
+            tag_name = tag['name']
+            # should retrieve all "en:blablabla, tag_name" or "it:tag_name"
+            tag_regex = re.compile(f'\n([a-z][a-z]:(?:[\w\s\-\']*\,-)*{tag_name})[,|\n]')
+
+            with open(file, "r") as f:
+
+                tag_regex_res = tag_regex.findall(f.read().lower().replace(" ", "-", -1))
+                print(tag_regex_res)
+
+                # found more than a single occurence in the taxonomy
+                # if exists, take value that correspond to "en" (i.e., unknown but 
+                #   already referenced in the taxonomy)
+                # otherwise (i.e., only different languages than "en"), keep first occurence
+                if len(tag_regex_res) > 1:
+                    # in the case that "en" is not in the list
+                    tag_regex_res_first = tag_regex_res[0]
+                    
+                    tag_regex_res = [x for x in tag_regex_res if "en:" in x]
+
+                    # "en" was not in the last put back first value in the list
+                    if not tag_regex_res:
+                        tag_regex_res.append(tag_regex_res_first)
+
+                # got one occurence in the taxonomy
+                if len(tag_regex_res) == 1:
+                    # world is in "en", hence if the tag is found in the taxonomy for "en" line,
+                    # it means that the tag is already referenced in the taxonomy
+                    if tag_regex_res[0][:2] == "en":
+                        already_referenced_tags.append(tag['id'])
+                    else:
+                        possible_wrong_language_tags[tag['id']] = tag_regex_res[0].split(',')[0]
+
+
+                # 0 occurences
+                else:
+                    possible_new_tags.append(tag['id'])
+
+
+    # print (for dev only) and save results of possible new tags in a file
+    if dev:
+        print(f"> Possible new tags for {tag_type}: <")
+        for possible_new_tag in possible_new_tags:
+            print("  ", possible_new_tag)
+
+        print(f"> Already referenced tags for {tag_type}: <")
+        for known_tag in already_referenced_tags:
+            print("  ", known_tag)
+
+        print(f"> Tags to update for {tag_type} (current => new): <")
+        for current_tag, updated_tag in possible_wrong_language_tags.items():
+            print(f"  {current_tag} => {updated_tag}")
+
+    # only save possible new tags to be reviewed and added
+    with open("update_tags_per_languages_possible_new_tags_" + tag_type, "w") as f:
+        f.write("possible_new_tags")
+    with open("update_tags_per_languages_possible_new_tags_" + tag_type, "a") as f:
+        for possible_new_tag in possible_new_tags:
+            f.write("\n" + possible_new_tag) 
```

Comment:
The search can be long, so I would save all the data you found out in a json file (for each map).

This way you could add an option to only load those json file and process them (if they exists), avoiding rerunning the search phase in case the update phase crashes.